addons/mail/models/ir_actions.py

```diff
@@ -22,6 +23,12 @@ class ServerActions(models.Model):
         'mail.template', 'Email Template', ondelete='set null',
         domain="[('model_id', '=', model_id)]",
     )
+    # Next Activity
+    activity_type_id = fields.Many2one('mail.activity.type', string='Next Activity Type')
```

Comment:
You should ensure that activity type is either generic (model False) or linked to the action's model.